src/Vault.sol

```diff
@@ -126,26 +126,26 @@ contract Vault is IVault, VaultToken, Ownable {
         SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
     }
 
-    function settleAndRefund(Currency currency, address to)
+    function settleAndMintRefund(Currency currency, address to)
         external
         payable
         override
         isLocked
         returns (uint256 paid, uint256 refund)
     {
-        paid = currency.balanceOfSelf() - reservesOfVault[currency];
-        int256 currentDelta = SettlementGuard.getCurrencyDelta(msg.sender, currency);
+        uint256 reservesBefore = reservesOfVault[currency];
+        reservesOfVault[currency] = currency.balanceOfSelf();
+        paid = reservesOfVault[currency] - reservesBefore;
 
+        int256 currentDelta = SettlementGuard.getCurrencyDelta(msg.sender, currency);
         if (currentDelta >= 0 && paid > currentDelta.toUint256()) {
```

Comment:
Can cache currentDelta after casting to uint256 so you don't have to keep casting it again, wasting gas.

```
        if (currentDelta >= 0) {
            uint256 currentDeltaUint256 = currentDelta.toUint256();
            if (paid > currentDeltaUint256 ) {
                // msg.sender owes vault but paid more than whats owed
                refund = paid - currentDeltaUint256;
                paid = currentDeltaUint256;
            }
        }
```

Reply:
Also just to confirm, refund is only done, and paid is only adjusted if currentDelta is positive or zero, never if currentDelta is negative?

Reply:
> Also just to confirm, refund is only done, and paid is only adjusted if currentDelta is positive or zero, never if currentDelta is negative?

yep. do you think theres a scenario where negative balanceDelta (vault owes caller token) might need to handled?

Reply:
Not sure, that's why I'm asking.